src/v2/Components/FlashBanner/index.tsx

```diff
@@ -1,51 +1,71 @@
 import React from "react"
-import { Banner, Flex, Sans } from "@artsy/palette"
+import { Banner, Button, Flex, Sans } from "@artsy/palette"
 import { QueryRenderer, graphql } from "react-relay"
 import { useSystemContext } from "v2/Artsy"
-import { ContentCode, messages, selectBannerMessage } from "./messages"
-import { BannerContentProps } from "./types"
+import qs from "qs"
+import { EmailConfirmationCTA } from "./EmailConfirmationCTA"
+import { BannerContent, BannerContentProps } from "./types"
 
 // import createLogger from "v2/Utils/logger"
 // const logger = createLogger("FlashBanner")
+interface FlashBannerProps {
+  contentCode?: string
+  me?: {
+    canRequestEmailConfirmation: boolean
+    email: string
+    id: string
+  }
+}
+
+function isComponent(
+  c: BannerContent
+): c is React.ComponentType<BannerContentProps> {
+  return typeof c === "function"
+}
 
 /**
  * The component responsible for selecting a flash message key from the `flash_message` url query param
  * `contentCode` prop is for storybooks and testing.
  */
-export const BaseFlashBanner: React.FC<BannerContentProps & {
-  contentCode?: ContentCode
-}> = props => {
-  const [contentCode, setContentCode] = React.useState<string | null>(
-    props.contentCode
-  )
-  const { relayEnvironment } = useSystemContext()
+export const BaseFlashBanner: React.FC<FlashBannerProps> = props => {
+  const contentMap: Partial<Record<string, any>> = {
+    /* Comes from actual query params */
+    confirmed: "Your email has been confirmed.",
+    already_confirmed: "You have already confirmed your email.",
+    invalid_token: "An error has occurred. Please contact support@artsy.net.",
+    blank_token: "An error has occurred. Please contact support@artsy.net.",
+    expired_token: "Link expired. Resend verification email.",
+
+    /* End of Query Params, beginning of constants that come from logic */
+    email_confirmation_cta: EmailConfirmationCTA,
+  }
 
-  React.useEffect(() => {
-    // TODO: verify this won't cause a nasty bug where any incoming messageCode prop would break the messageCode logic
-    if (!contentCode) {
-      const newCode = selectBannerMessage(props.me, window.location)
+  let { contentCode } = props
 
-      setContentCode(newCode)
+  if (!contentCode) {
+    const query = qs.parse(window.location.search.slice(1))
+    contentCode = query["flash_message"]
+  }
+
+  if (!contentCode) {
+    if (props.me?.canRequestEmailConfirmation) {
+      contentCode = "email_confirmation_cta"
     }
-  }, [props.me, contentCode])
+  }
+
+  const Content = contentMap[contentCode]
```

Comment:
This is looking much better. 

Rather than introducing branching logic and helper functions around whether or not something is a string or a component (two very different data structures) it would be good to look at whats needed here, meaning: 
- If the content code is `email_confirmation_cta`, that means we need a component. We know this ahead of time. So to disambiguate, why not something like:
```jsx
<Flex justifyContent="center" alignItems="center">
  {contentCode === "email_confirmation_cta" 
    ? < EmailConfirmationCTA me={me} />
    : contentMap[contentCode]}
</Flex>
```